Column clearing changes

rchinerman opened the pull request:
This PR contains a bug fix that stopped the clear/unclear logic inside of the column header from working. It also contains multiple icon changes for increased clarity regarding what the button does, and a wording improvement.

brunolemos commented:
Thanks! I don't really plan to add the icon and color changes, it's a bit distracting. I'll follow something similar to TweetDeck's design.

> bug fix that stopped the clear/unclear logic inside of the column header from working

Can you please clarify? How can I reproduce it? I see now the Clear button appears when the Archived filter is enabled but I don't think that's the desirable behavior. The "Clear" is basically the same as "Archive" so you shouldn't be able to archive what's already archived.

rchinerman commented:
@brunolemos There was existing logic to switch between clear/unclear in the column header, but the overall ternary was causing 'clear' to never show up. It would either say 'unclear' or disappear entirely. I put parentheses around the logic, but then Prettier cleaned it up and removed some parentheses for the same effect.

No coloring is fine, I just added it while I was in there because 'unclear' was colored but nothing else would have been.

I do think the icons offer value, much more accurately reflecting what the button's purpose is, but it's really not a big change at the end of the day.

I like the idea of copying TweetDeck's implementation, I think removing Archive would be fine.
